### PR TITLE
Implements RTK Query

### PR DESCRIPTION
**Pull request by scandycuz**
### What

Implements RTK Query in the app for songs (this still won't work as the authorization isn't set up, but this initial implementation should provide a starter template for implementing other API calls such as registration)

### Note

Since this doesn't actually add any app functionality, it shouldn't need to be tested

### CODE REVIEW

**Comment by j-mateo** [comment]
Task linked: [CU-2jku1e3 Implement RTK Query (to enable making API calls)](https://app.clickup.com/t/2jku1e3)

File: src/api/api.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+const api = createApi({
+  reducerPath: "api",
+  baseQuery: fetchBaseQuery({
+    baseUrl: "https://staging-newm-server.herokuapp.com/v1/",
+  }),
+  endpoints: () => ({}),
+});
+
```
Comment (scandycuz) [inline src/api/api.ts:10]: This sets up the base url for the NEWM API

File: src/modules/song/api.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+const extendedApi = api.injectEndpoints({
+  endpoints: (build) => ({
+    getSongs: build.query({
+      query: () => "portal/songs",
+    }),
+  }),
+  overrideExisting: false,
+});
+
```
Comment (scandycuz) [inline src/modules/song/api.ts:11]: Individual endpoints can be added in each module by extending the base API

File: src/modules/song/slice.ts [hunk 3]
```diff
@@ -1,264 +1,27 @@
...
   name: "role",
   reducers: {},
+  extraReducers: (builder) => {
+    builder.addMatcher(
+      api.endpoints.getSongs.matchFulfilled,
+      (state, { payload }) => {
+        state.songs = payload.songs;
+      }
+    );
+  },
```
Comment (scandycuz) [inline src/modules/song/slice.ts:24]: These matchers will update the Redux store when specific queries are completed

**Review by HenryQuillin — APPROVED** [review]
LGTM

**Pull request merged**